src/app.js

```diff
@@ -4,13 +4,25 @@ const bodyParser = require("body-parser");
 const cors = require("cors");
 const controllers = require("./controllers");
 const { errors } = require("celebrate");
+const { whitelist } = require("../lib/config")
+
+var corsOptions = {
+  origin: function (origin, next) {
+    if (whitelist.indexOf(origin) !== -1) {
+      next(null, true);
+    } else {
+      next(new Error('Not allowed by CORS'));
+    }
+  }
+}
 
 const app = express();
 app.use(helmet());
 app.use(cors());
 app.use(bodyParser.urlencoded({ extended: true }));
 app.use(bodyParser.json());
-app.use("/api/v1/", controllers);
+app.use(function(req,res,next){ req.headers.origin = req.headers.origin || req.headers.host; next(); })
```

Comment:
Hi @mehtankush what does this do?

Reply:
As seen in the below req.headers object, the request from my localhost doesn't have an `origin` property but has the `host` value. I copied this to the `origin` to handle all the scenarios.

```
req.headers:  { host: 'localhost:3000',
  connection: 'keep-alive',
  pragma: 'no-cache',
  'cache-control': 'no-cache',
  'upgrade-insecure-requests': '1',
  'user-agent': 'Mozilla/5.0 (Macintosh; Intel Mac OS X 10_13_4) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/69.0.3497.100 Safari/537.36',
  accept: 'text/html,application/xhtml+xml,application/xml;q=0.9,image/webp,image/apng,*/*;q=0.8',
  'accept-encoding': 'gzip, deflate, br',
  'accept-language': 'en-GB,en-US;q=0.9,en;q=0.8',
  cookie: '_ga=GA1.1.1360559728.1534965181; gsScrollPos-1144579231=0; gsScrollPos-1144581659=0; io=zPFb_F8GTUXljV9DAAAA' }
```

Reply:
@mehtankush this is because you are on localhost. Host is the server, origin is the client. We want to check the origin against the whitelist, not the host. 
I just added a config property `isProduction` to the config.
Can you please use a conditional statement to check isProduction when setting the cors?
Like this:
```
if(isProduction) app.use(cors(corsOptions));
```
That way we do not have to worry about cors when we are testing or developing.
Thanks.

Reply:
Please merge the latest changes from the master branch into your branch. Then you will see the updated code I have added with the config properties.